ocs_ci/utility/perf_dash/dashboard_api.py

```diff
@@ -0,0 +1,510 @@
+
+        Args:
+            version (int): The version ID
+
+        Returns:
+             str : the version name in the DB
+
+        """
+        return self.get_name_by_id("versions", version)
+
+    def get_test_name(self, test=0):
+        """
+        Query of the Name of test ID in the DB
+
+        Args:
+            test (int): The test ID
+
+        Returns:
+             str : the Test Name in the DB
+
+        """
+        return self.get_name_by_id("tests", test)
+
+    def get_platform_name(self, platform=0):
+        """
+        Query of the IName of platform ID in the DB
+
+        Args:
+            platform (int): The platform ID
+
+        Returns:
+             str : the Platform Name in the DB
+
+        """
+        return self.get_name_by_id("platform", platform)
+
+    def get_topology_name(self, topology=0):
+        """
+        Query of the Name of platform ID in the DB
+
+        Args:
+            topology (int): The Topology ID
+
+        Returns:
+             str : the Topology Name in the DB
+
+        """
+        return self.get_name_by_id("az_topology", topology)
+
+    def get_version_builds(self, version=None):
+        """
+        Query the list of build in specific version
+
+        Args:
+            version (str): The version name (e.g. 4.9.0)
+
+        Returns:
+             dict : dictionary of (Name: ID), None if not exist
+
+        """
+        ver_id = self.get_version_id(version)
+        if ver_id is None:
+            return None
+
+        results = {}
+        query = f"SELECT id, name FROM builds WHERE version = {ver_id} ;"
+        if self._run_query(query=query):
+            for (build_id, name) in self.cursor:
+                results[name] = build_id
+
+        return None if results == {} else results
+
+    def get_build_id(self, version, build):
+        """
+        Getting the build ID for specific build of version.
+        if the build is not exist, return None
+
+        Args:
+            version (str): the version name (e.g. 4.9.0)
+            build (str): the build name (e.g. GA)
+
+        Returns:
+            int : the build ID
+
+        """
+        all_builds = self.get_version_builds(version=version)
+        if all_builds:
+            return all_builds.get(build)
+        return None
+
+    def insert_build(self, version, build):
+        """
+        Insert a new build to the DB and return it's ID
+
+        Args:
+            version (str): The version number as string (e.g. 4.9.0)
+            build (str): The build number (e.g. 180 / RC1-200 / GA)
+
+        Returns:
+             int : the ID of the build in the DB, otherwise None
+
+        """
+        # prevent of duplicate records for the same build
+        build_id = self.get_build_id(version=version, build=build)
+        if build_id is not None:
+            return build_id
+
+        # Try to insert the version into the DB, it will not inserted twice,
```

Comment:
spelling suggestiong " Try to insert the version into the DB, it will not be inserted twice"

Reply:
done.